python/ray/serve/_private/application_state.py

```diff
@@ -908,6 +910,10 @@ def build_serve_application(
         deploy_args_list = []
         for deployment in deployments:
             is_ingress = deployment.name == ingress.name
+            # If app log config is set and deployment config is not set,
+            # set app config to deployment log config.
+            if deployment.logging_config is None and logging_config:
+                deployment.set_logging_config(logging_config.dict())
```

Comment:
Changing logging config doesn't change the code version right? If you set the deployment's logging config to the application logging config here, how can you distinguish in `override_deployment_info` between a logging config that was set at the deployment level versus a logging config that was inherited from the app level? After `build_serve_application` is run, seems like every deployment will have a logging config (so no overriding will happen in `override_deployment_info`?)

Reply:
good catch, i removed the logic, relying on the override_deployment_info to do the trick.